kos_sim/eval_model.py

```diff
@@ -0,0 +1,352 @@
+"""Real robot walking deployment script using PyKOS interface."""
+
+import argparse
+import asyncio
+import math
+import time
+from copy import deepcopy
+from typing import Dict, List, Tuple, Union
+import csv
+
+import numpy as np
+import pygame
+from kinfer.inference.python import ONNXModel
+import pykos
+from pykos import KOS
+from scipy.spatial.transform import Rotation as R
+
+ARM_IDS = [11, 12, 13, 14, 15, 16, 21, 22, 23, 24, 25, 26]
+
+upper_arm_ids = [11, 12, 13, 14, 15, 21, 22, 23, 24, 25]
+
+# Ordered joint names for policy (legs only, top-down, left-right order)
+JOINT_NAME_LIST = [
+    "L_hip_y",    # Left leg, top to bottom
+    "L_hip_x",
+    "L_hip_z", 
+    "L_knee",
+    "L_ankle",
+    "R_hip_y",    # Right leg, top to bottom
+    "R_hip_x",
+    "R_hip_z",
+    "R_knee",
+    "R_ankle"
+]
+
+# Joint mapping for KOS
+JOINT_NAME_TO_ID = {
+    # Left leg
+    "L_hip_y": 31,
+    "L_hip_x": 32,
+    "L_hip_z": 33,
+    "L_knee": 34,
+    "L_ankle": 35,
+    # Right leg
+    "R_hip_y": 41,
+    "R_hip_x": 42,
+    "R_hip_z": 43,
+    "R_knee": 44,
+    "R_ankle": 45
+}
+
+# Joint signs for correct motion direction
+JOINT_SIGNS = {
+    # Left leg
+    "L_hip_y": 1,
+    "L_hip_x": 1,
+    "L_hip_z": 1,
+    "L_knee": -1,
+    "L_ankle": 1,
+    # Right leg
+    "R_hip_y": 1,
+    "R_hip_x": 1,
+    "R_hip_z": 1,
+    "R_knee": 1,
+    "R_ankle": -1
+}
+
+def handle_keyboard_input() -> None:
+    """Handle keyboard input for velocity commands."""
+    global x_vel_cmd, y_vel_cmd, yaw_vel_cmd
```

Comment:
prefer not using globals, better to refactor this into a class and use class vars instead